ls1intum/Artemis
# User-Statistics/tests

### Checklist


### Motivation and Context


- User statistics was not yet tested
- Service method convertMapList was too complex

### Description

- Removed a method not used
- Pushed client tests for the user statistics page up to 92%
- Improved server side tests
- Extracted a functionality and added a helper class for the convertMapList method in StatisticsService

### Steps for Testing


1. Log in to Artemis
2. Navigate to Server Administration -> User statistics
3. Make sure nothing is broken when changing tabs and the active user/ logged-in users and active tutors graphs are correct/make sense (other graphs were not changed)

### Test Coverage



statistics.component.spec.ts: 100%
statistics.service.ts: 100%
statistics-graph.component.spec.ts: 91%
StatisticsService.java : 93%
StatisticsResource.java: still 100%

Exam testing:
exercise-group-update.component.ts: 100%
